### PR TITLE
fix(amplify-provider-awscloudformation): Fix amplify configure opening wrong IAM User Creation URL in Windows & WSL

### LINKED ISSUE #6836: Configure opens wrong URL to create IAM from tmux
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm -g

### If applicable, what version of Node.js are you using?

v14.15.1

### Amplify CLI Version

4.45.0

### What operating system are you using?

Ubuntu 20.04 (WSL2 on Windows10)

### Amplify Categories

Not applicable

### Amplify Commands

configure project

### Describe the bug

`amplify configure` asks a new IAM user name, then open the URL for that. The URL in browser is wrong while the URL printed in console is correct.

e.g.

- `https://console.aws.amazon.com/iam/home?region=us-west-2#/users=final&accessKey&userNames=xxxxxxxx&permissionType=policies&policies=arn:aws:iam::aws:policy%2FAdministratorAccess` (wrong, in browser, shows IAM home instead)
- `https://console.aws.amazon.com/iam/home?region=us-west-2#/users$new?step=final&accessKey&userNames=xxxxxxxx&permissionType=policies&policies=arn:aws:iam::aws:policy%2FAdministratorAccess` (correct, in console)

For short, `#/users=final` vs `#/users$new?step=final`.

### Expected behavior

Open the correct URL so that user can create it quickly

### Reproduction steps

1. Move to the project directory in tmux
1. Run `amplify configure`
2. Proceed to "Specify the username of the new IAM user"
3. Input any username, and hit enter

### Log output

```
# Put your logs below this line

```

- The browser I use is Edge
- If ran `amplify configure` directly on WSL console, it does not open at all. **This happens only with tmux**
- tmux: https://github.com/tmux/tmux

### PR DESCRIPTION
**Pull request by rajrajhans**
*Issue #, if available:* Closes #6836 and #5664 

*Description of changes:*
Escapes $ in IAM user creation URL string for Windows & WSL environments, since powershell treats it as environment var and therefore ignores it.

_More details_:
- Checks whether the platform is windows or WSL.
    - using `process.platform` to check for windows.
    - using an existing dependency `is-wsl` ( `open` depends on `is-wsl` ) to check for wsl.
- if it is either windows or wsl, $new is replaced by \`$new in deepLinkURL. on line 53, deepLinkURL is printed without ` to ensure that correct url is shown in the terminal.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/setup-new-user.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ const constants = require('./constants.js');
 const systemConfigManager = require('./system-config-manager');
 const obfuscationUtil = require('./utility-obfuscate');
 const { open } = require('amplify-cli-core');
+const isOnWsl = require('is-wsl');
```
Comment (jhockett) [inline packages/amplify-provider-awscloudformation/src/setup-new-user.js:9]: This dependency needs to be added to `packages/amplify-provider-awscloudformation/package.json`

File: packages/amplify-provider-awscloudformation/src/setup-new-user.js [hunk 2]
```diff
@@ -43,9 +44,13 @@ async function run(context) {
     },
   ]);
 
-  const deepLinkURL = constants.AWSCreateIAMUsersUrl.replace('{userName}', userName).replace('{region}', awsConfig.region);
+  let deepLinkURL = constants.AWSCreateIAMUsersUrl.replace('{userName}', userName).replace('{region}', awsConfig.region);
+  const isOnWindows = /^win/.test(process.platform);
```
Comment (cjihrig) [inline packages/amplify-provider-awscloudformation/src/setup-new-user.js:48]: It's safe to use `process.platform === 'win32'` for these checks.

Comment (rajrajhans) [reply packages/amplify-provider-awscloudformation/src/setup-new-user.js]: Done. Thanks.

File: packages/amplify-provider-awscloudformation/src/setup-new-user.js (hunk 1 above)
Comment (rajrajhans) [reply packages/amplify-provider-awscloudformation/src/setup-new-user.js:9]: Added. Thanks for the review.

**Review by jhockett — APPROVED** [review]
LGTM

**Pull request merged**